### PR TITLE
Feature/crum 161 order service authorization

### PR DESCRIPTION
**Pull request by elijahbrooks**

### CODE REVIEW

File: src/main/java/com/crumbs/orderservice/controller/MainController.java [hunk 1]
```diff
@@ -28,17 +29,19 @@
...
             @RequestParam(defaultValue="3") Integer size
     ){
-        OrdersDTO orders = orderService.getOrdersDTO(id, PageRequest.of(page, size, Sort.by(Sort.Direction.DESC,
-                "createdAt")));
+        OrdersDTO orders = orderService.getOrdersDTO(username,
+                PageRequest.of(page, size, Sort.by(Sort.Direction.DESC, "createdAt")));
         return new ResponseEntity<>(orders, HttpStatus.OK);
     }
 
+    @PreAuthorize("permitAll()")
```
Comment (jim-brower) [inline src/main/java/com/crumbs/orderservice/controller/MainController.java]: admin only

File: src/main/java/com/crumbs/orderservice/controller/MainController.java [hunk 2]
```diff
@@ -55,106 +58,133 @@
...
+        List<CartItem> cartItems = cartService.removeItem(username, cartId);
         return new ResponseEntity<>(cartItems, HttpStatus.OK);
     }
 
+    @PreAuthorize("hasAuthority('ADMIN') or (hasAuthority('CUSTOMER') and #username == authentication.principal)")
     @DeleteMapping("orders/{id}")
     public ResponseEntity<Object> deleteOrder(
-            @PathVariable Long id
+            @PathVariable Long id,
+            @RequestHeader("Username") String username
```
Comment (jim-brower) [inline src/main/java/com/crumbs/orderservice/controller/MainController.java]: make header optional

**Review by JonathanFrey0000 — APPROVED** [review]
looks good

**Review by jim-brower — APPROVED** [review]
good job

**Pull request merged**